### PR TITLE
Sync berry crush from Emerald

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/berry_crush.c [hunk 1]
```diff
@@ -800,172 +978,173 @@ u32 ExitBerryCrushWithCallback(MainCallback callback)
...
+
+    switch (sGame->playerCount)
     {
     case 2:
-        if (sBerryCrushGamePtr->pressingSpeed > gSaveBlock2Ptr->berryCrush.berryCrushResults[0])
+        if (sGame->pressingSpeed > gSaveBlock2Ptr->berryCrush.berryCrushResults[0])
         {
-            sBerryCrushGamePtr->unk25_1 = 1;
-            gSaveBlock2Ptr->berryCrush.berryCrushResults[0] = sBerryCrushGamePtr->pressingSpeed;
+            sGame->newRecord = 1;
```
Comment (GriffinRichards) [inline src/berry_crush.c:1078]: `TRUE`, and below

File: include/math_util.h [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-s32 MathUtil_Div32(s32 x, s32 y);
-s32 MathUtil_Mul32(s32 x, s32 y);
-s16 MathUtil_Div16Shift(u8 s, s16 x, s16 y);
-s16 MathUtil_Mul16Shift(u8 s, s16 x, s16 y);
+s16 invfx16(s16 y);
+s16 mulfx16(s16 x, s16 y);
+s32 divfx32(s32 x, s32 y);
+s32 mulfx32(s32 x, s32 y);
+s16 divfxn16(u8 s, s16 x, s16 y);
+s16 mulfxn16(u8 s, s16 x, s16 y);
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/math_util.h:9]: fx == fixed point rational. notation mirrors the fx types on nds

[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/math_util.h:9]: to further clarify, fxn is a working notation for `fixed point with arbitrary precision n`

[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/math_util.h]: I renamed the math_util functions again to sync up with the macros in global.h

[the remark this answers is not in the document]
Comment (GriffinRichards) [reply include/math_util.h]: The names are still being used in the typedefs that you've since added, which also seem unnecessary

**Pull request merged**